operator/src/main/java/org/bf2/operator/managers/ResourceManager.java

```diff
@@ -0,0 +1,339 @@
+        if (update) {
+            configMap = new ConfigMapBuilder(configMap).withData(newData).build();
+            client.configMaps()
+                    .withName(FLEETSHARD_RESOURCES)
+                    .lockResourceVersion(configMap.getMetadata().getResourceVersion())
+                    .replace(configMap);
+        }
+        return configMap;
+    }
+
+    static Resources createResources(ManagedKafka mk, String currentProfile) {
+        Resources resources = new Resources();
+        resources.profile = currentProfile;
+        resources.units = Integer.parseInt(
+                OperandUtils.getOrDefault(mk.getMetadata().getLabels(), ManagedKafka.PROFILE_QUOTA_CONSUMED, "1"));
+        return resources;
+    }
+
+    public synchronized void releaseResources(ManagedKafka managedKafka) {
+        // use the latest - TODO: see if using the cached version is acceptable
+        ConfigMap resourceConfigMap = client.configMaps().withName(FLEETSHARD_RESOURCES).get();
+        if (resourceConfigMap != null) {
+            String entryKey = Cache.metaNamespaceKeyFunc(managedKafka);
+            if (releaseResources(resourceConfigMap.getData(), entryKey)) {
+                client.configMaps()
+                        .withName(FLEETSHARD_RESOURCES)
+                        .lockResourceVersion(resourceConfigMap.getMetadata().getResourceVersion())
+                        .replace(resourceConfigMap);
+            }
+        }
+    }
+
+    private boolean releaseResources(Map<String, String> resourceData, String entryKey) {
+        String previous = resourceData.remove(entryKey);
+        if (previous != null) {
+            Resources resources = Serialization.unmarshal(previous, Resources.class);
+            resourceData.computeIfPresent(resources.profile,
+                    (k, v) -> String.valueOf(Integer.parseInt(v) - resources.units));
+            return true;
+        }
+        return false;
+    }
+
+    public synchronized Optional<OperandReadiness> claimResources(ManagedKafka managedKafka, String profile,
+            ManagedKafkaAgent agent) {
+        Integer max = getProfileMaxUnits(agent.getSpec().getCapacity(), profile).orElse(null);
+        if (max == null) {
+            return Optional.empty();
+        }
+
+        // ensure we have the latest
+        ConfigMap resourceConfigMap = client.configMaps().withName(FLEETSHARD_RESOURCES).get();
+
+        Resources resources = createResources(managedKafka, profile);
+
+        int total = Integer.parseInt(resourceConfigMap.getData().getOrDefault(profile, "0"));
+
+        int newTotal = resources.units + total;
+        if (newTotal > max) {
+            log.warnf("Rejecting the deployment of %s/%s as the cluster is full",
+                    managedKafka.getMetadata().getNamespace(), managedKafka.getMetadata().getName());
+            return Optional
+                    .of(new OperandReadiness(Status.False, Reason.Rejected, "Cluster has insufficient resources"));
```

Comment:
@VanillaSpoon this is the status that you'll be interested in intepreting:
- Status: `Ready`, `False`
- Reason: `Rejected`
- Message: `Cluster has insufficient resources`

Reply:
Thanks @machi1990